tests/jinja_context_addons_tests.py

```diff
@@ -0,0 +1,140 @@
+        self.assertEqual(" AND ((src_num_ip >= 50331648 AND src_num_ip <= 67108863) OR (src_num_ip = 33686018) OR (dst_num_ip >= 50331648 AND dst_num_ip <= 67108863) OR (dst_num_ip = 33686018))", rendered)
+
+
+    def test_render_ipv4_between_number_colums_template(self) -> None:
+        test_filter = [
+            {
+                "col": "src_ip_num",
+                "op": "LIKE",
+                "val": ['12.0.0.0/8', '2.0.0.0/16']
+            }
+        ]
+        rendered = render_ipv4_between_number_colums(test_filter, '1.1.1.1', '2.2.2.2' )
+        self.assertEqual(
+            ''' AND (( (1.1.1.1 <= 201326592 AND 2.2.2.2 >= 201326592)
+                    OR (1.1.1.1 <= 218103807 AND 2.2.2.2 >= 218103807)
+                    OR (201326592 <= 1.1.1.1 AND 2.2.2.2 <= 218103807) ) OR ( (1.1.1.1 <= 33554432 AND 2.2.2.2 >= 33554432)
+                    OR (1.1.1.1 <= 33619967 AND 2.2.2.2 >= 33619967)
+                    OR (33554432 <= 1.1.1.1 AND 2.2.2.2 <= 33619967) ))'''
+        , rendered)
+
+
+    def test_render_in_conditions_template(self) -> None:
+        test_ip_array=['1.1.1.1','240.0.0.0/4']
+        rendered = render_in_conditions(test_ip_array, "src_num_ip")
+        self.assertEqual(['(src_num_ip = 16843009)', '(src_num_ip >= 4026531840 AND src_num_ip <= 4294967295)'], rendered)
+
+    def test_dashobard_link_template(self) -> None:
+        test_link_label = "LABEL"
+        test_dashboard_id = 2301
+        test_src_column = 'test_col'
+        test_target_column = 'target_col'
+
+        rendered = dashobard_link(test_link_label, test_dashboard_id, test_src_column, test_target_column)
+        self.assertEqual(" concat('<a href=\"http://10.162.232.22:8088/superset/dashboard/2301/?preselect_filters={%22160%22:{%22target_col%22:[%22', target_col, '%22]}}\">LABEL</a>' ) ", rendered)
+
+    #Test for Exceptions
+
+    def test_ipv4str_to_number_template_exception(self) -> None:
+        self.assertRaises(OSError, ipv4str_to_number, '1912.168.0.0')
+
+    def test_render_ipv4_column_template_exception(self) -> None:
+        test_filter = [
+            {
+                "col": "src_ip_num",
+                "op": "==",
+                "val": "1.1.1.1"
+            },
+            {
+                "col": "src_ip_num",
+                "op": "IN",
+                "val": ['3.3.3.3/8', '2.2.2.2']
+            }
+        ]
+        self.assertRaises(ValueError, render_ipv4_number_column, test_filter, 'src_ip_num' )
+
+    def test_render_ipv4_either_number_columns_template_exception(self) -> None:
+        test_filter = [
+            {
+                "col": "src_ip_num",
+                "op": "==",
+                "val": ['3.0.0.0/8', '2.2.2.200/34']
+            }
+        ]
+        self.assertRaises(NetmaskValueError, render_ipv4_either_number_columns, test_filter, "src_num_ip", "dst_num_ip")
+
+    def test_render_ipv4_between_number_colums_template_exception(self) -> None:
```

Comment:
```suggestion
    def test_render_ipv4_between_number_colums_template_invalid_operator(self) -> None:
```
Same comment for the other method names as well. If we can specify what exception we're testing, it might make the tests more easily readable. 

Also, perhaps we should be testing each possible exception for each method? e.g., check for an exception for an invalid operator from each render_* method.

Reply:
I looked into this a bit more and it seems like having an invalid operator does not produce any errors

Reply:
Should it, perhaps? If so, maybe raise a ticket to fix that. It's probably a bit out of scope for your testing task.

Reply:
No I don't think the filters will ever get an invalid choice, since you have to pick between the 7/8 different options superset ui gives you